include/AnnotationHandler.h

```diff
@@ -0,0 +1,42 @@
+#ifndef ANNOTATION_HANDLER_H
+#define ANNOTATION_HANDLER_H
+
+#include "Annotation.h"
+#include "Debug.h"
+#include "ErrorAnnotation.h"
+#include "FunctionAnnotation.h"
+#include "ParameterAnnotation.h"
+#include "ReturnAnnotation.h"
+#include "StructAnnotation.h"
+#include "Utils.h"
+
+class AnnotationHandler {
+private:
+  std::list<Annotation *> annotations;
+
+public:
+  AnnotationHandler();
+  AnnotationHandler(const std::string &fileName);
+  void addAnnotation(const std::string &rawAnnotationString);
+  void addAnnotationsFromFile(const std::string &fileName);
+
+  // returns Annotation* instead of more specific type (in this case,
+  // FunctionAnnotation*) since we may not find the annotation and will want to
+  // return ErrorAnnotation* instead
```

Comment:
C++ has multiple inheritance, so we could also fix this issue by changing the type hierarchy. In particular, we could add types like `FunctionErrorAnnotation` that are both a `FunctionAnnotation` and an `ErrorAnnotation`; that is, make the type hierarchy look like this:
```
         Annotation
       /                   \
ErrorAnno        FunctionAnno
      \                     /
    FunctionErrorAnno
```

(and then we'd do the same for parameter annotations, etc.)

I'm not sure that this is worth it, though.

Reply:
In that case I'd suggest adding a todo comment (maybe just copy what I wrote above?)